uxarray/grid/grid.py

```diff
@@ -1378,3 +1442,241 @@ def isel(self, **dim_kwargs):
+                )
+                final_faces.append(_face)
+
+                # node_zero = node_0 - node_central
+                #
+                # node_cross = np.cross(node_0, node_central)
+                # node_zero_mag = np.linalg.norm(node_zero)
+                #
+                # d_angles = np.zeros(n_edges, dtype=np.float64)
+                # d_angles[0] = 0.0
+                # for j in range(1, n_edges):
+                #     if face_temp[j] is not None:
+                #         sub = np.array(
+                #             [
+                #                 dual_node_x[face_temp[j]],
+                #                 dual_node_y[face_temp[j]],
+                #                 dual_node_z[face_temp[j]],
+                #             ]
+                #         )
+                #         node_diff = sub - node_central
+                #         node_diff_mag = np.linalg.norm(node_diff)
+                #
+                #         d_side = np.dot(node_cross, node_diff)
+                #         d_dot_norm = np.dot(node_zero, node_diff) / (
+                #             node_zero_mag * node_diff_mag
+                #         )
+                #
+                #         if d_dot_norm > 1.0:
+                #             d_dot_norm = 1.0
+                #
+                #         d_angles[j] = np.arccos(d_dot_norm)
+                #
+                #         if d_side > 0.0:
+                #             d_angles[j] = -d_angles[j] + 2.0 * np.pi
+                #
+                # d_current_angle = 0.0
+                # face.set_node(0, face_temp[0])
+                #
+                # for j in range(1, n_edges):
+                #     ix_next_node = -1
+                #     d_next_angle = 2.0 * np.pi
+                #
+                #     for k in range(1, n_edges):
+                #         if d_current_angle < d_angles[k] < d_next_angle:
+                #             ix_next_node = k
+                #             d_next_angle = d_angles[k]
+                #
+                #     if ix_next_node == -1:
+                #         continue
+                #
+                #     face.set_node(j, face_temp[ix_next_node])
+                #     d_current_angle = d_next_angle
+                # final_faces.append(face.edges)
+
+        # Empty array to hold `node_face_connectivity`
+        node_face_connectivity = np.full(
+            (len(final_faces), self.n_max_node_faces),
+            INT_FILL_VALUE,
+            dtype=ux.INT_DTYPE,
+        )
+
+        # Populate the node_face array
+        for i, face in enumerate(final_faces):
+            faces = [
+                edge[0] for edge in face
+            ]  # Extract the first element of each edge tuple
+            node_face_connectivity[i, : len(faces)] = (
+                faces  # Assign the faces to the node_face array
+            )
+
+        # Construct dual mesh
+        dual = self.from_topology(
```

Comment:
Also, don't know if this would cause a problem but might want to change it back to ux.Grid.from_topology